Add/641 naked browser launch opts

inancgumus opened the pull request:
Fixes #641.

ankur22 commented on chromium/browser_type.go:
```diff
@@ -124,16 +124,15 @@ func (b *BrowserType) initContext() context.Context {
 // Launch allocates a new Chrome browser process and returns a new api.Browser value,
 // which can be used for controlling the Chrome browser.
-func (b *BrowserType) Launch(opts goja.Value) api.Browser {
+func (b *BrowserType) Launch(opts ...goja.Value) api.Browser {
```
Why does it need to be done in this way?

For example `browser.newContext()` can be used without supplying any options and it only takes a single `goga.Value` argument:

```go
func (b *Browser) NewContext(opts goja.Value) api.BrowserContext
```

ankur22 replied:
The browser context parse returns `ni`l if `opts` is `nil`:

```go
if opts != nil && !goja.IsUndefined(opts) && !goja.IsNull(opts) {
   ...
}
return nil
```

Couldn't we do something similar?

inancgumus replied:
That's a very good point 👍  I'll check it out, thanks.

ankur22 commented on examples/launch_naked.js:
```diff
@@ -0,0 +1,15 @@
+import exec from 'k6/execution';
```
Nice, this is good 👍

ankur22 commented on common/browser_options_test.go:
```diff
@@ -262,4 +263,41 @@ func TestBrowserLaunchOptionsParse(t *testing.T) {
 			}
 		})
 	}
+
```
Thanks for the test too 👍

ankur22 commented on common/browser_options_test.go:
```diff
@@ -262,4 +263,41 @@ func TestBrowserLaunchOptionsParse(t *testing.T) {
 	}
+
+	// will be passed to NewLaunchOptions to set onCloud to false.
+	const onCloud = false
```
Very minor, so you can ignore this suggestion.

```suggestion
	const notOnCloud = false
```

I think this reads better when it's used:

```go
wantOpts: NewLaunchOptions(notOnCloud),
```

vs

```go
wantOpts: NewLaunchOptions(onCloud),
```

ankur22 replied:
Or maybe set `onCloud` to `true`, and then:

```go
wantOpts: NewLaunchOptions(!onCloud),
```

ankur22 reviewed (changes requested):
I'm a little confused as to why we need to use a variadic function.

The tests are a great addition 👍

inancgumus commented:
You're right. IDK how I missed that 🤦 Thanks, @ankur22 🙇 I've changed the signature to the original and updated the tests. I only left-shifted the logger as I wanted to clarify that `context` and `logger` are peripherals, and we're interested in the options.

ankur22 reviewed (approved):
Looks good, thanks for this!
